src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -77,11 +78,11 @@ public static void generateReposReport(List<RepoConfiguration> configs, String o
         Map<RepoLocation, List<RepoConfiguration>> repoLocationMap = groupConfigsByRepoLocation(configs);
         cloneAndAnalyzeRepos(repoLocationMap, outputPath);
 
-        Date sinceDate = cliSinceDate == null ? earliestSinceDate : cliSinceDate;
-        Date untilDate = cliUntilDate == null ? latestUntilDate : cliUntilDate;
+        Date reportSinceDate = (cliSinceDate.equals(SinceDateArgumentType.ARBITARY_FIRST_COMMIT_DATE))
+                ? earliestSinceDate : cliSinceDate;
```

Comment:
would this conditional ever be true?
Also seems like the variable on like 76 is unused.

Reply:
Yes, if user sets `-s *`, `ARBITARY_FIRST_COMMIT_DATE` will be set to `cliSinceDate` first. Then after the repos are analyzed, `reportSinceDate` will be set to `earliestSinceDate`.

Removed the unused variable and methods.